keras/callbacks.py

```diff
@@ -468,12 +469,31 @@ def _set_model(self, model):
         self.model = model
         self.sess = KTF.get_session()
         if self.histogram_freq and self.merged is None:
-            layers = self.model.layers
-            for layer in layers:
-                if hasattr(layer, 'W'):
-                    tf.histogram_summary('{}_W'.format(layer.name), layer.W)
-                if hasattr(layer, 'b'):
-                    tf.histogram_summary('{}_b'.format(layer.name), layer.b)
+            for layer in self.model.layers:
+
+                weights = []
+                if hasattr(layer, 'trainable_weights'):
+                    weights.extend(layer.trainable_weights)
+                if hasattr(layer, 'non_trainable_weights'):
+                    weights.extend(layer.non_trainable_weights)
```

Comment:
The paragraph above can be replaced with `layer.weights`, which is always implemented.